src/samplers/support/adapt.jl

```diff
@@ -1,32 +1,58 @@
 # Acknowledgement: this adaption settings is mimicing Stan's 3-phase adaptation.
 
+# Ref： https://github.com/stan-dev/math/blob/develop/stan/math/prim/mat/fun/welford_var_estimator.hpp
+mutable struct VarEstimator{T<:Real}
+  n :: Int
+  μ :: Vector{T}
+  M :: Vector{T}
+end
+
+reset!(ve::VarEstimator) = begin
+  ve.n = 0
+  ve.μ = zeros(ve.μ)
+  ve.M = zeros(ve.M)
+end
+
+add_sample!{T<:Real}(ve::VarEstimator{T}, s::Vector{T}) = begin
+  ve.n += 1
+  δ = s - ve.μ
+  ve.μ += δ / ve.n
+  ve.M += δ .* (s - ve.μ)
+end
+
+get_var(ve::VarEstimator) = begin
+  @assert ve.n >= 2
+  var = ve.M / (ve.n - 1)
+  var = (ve.n / (ve.n + 5.0)) * var + 1e-3 * (5.0 / (ve.n + 5.0))
+  return var
+end
+
+
+
 type WarmUpManager
-  iter_n    ::    Int
-  state     ::    Int
+  adapt_n   ::    Int
   params    ::    Dict
```

Comment:
Use a more concrete type for `params :: Dict` if possible to improve efficiency.